scripts/utils/price_utils.js

```diff
@@ -179,11 +179,11 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
    * @param {integer} stableTokenDecimals number of decimals of the stable token
    * @return {BN[2]} amounts of stable token and target token for an unlimited order at the input price
    */
-  const getUnlimitedOrderAmounts = function (price, targetTokenDecimals, stableTokenDecimals) {
-    let targetTokenAmount = max128.clone()
+  const getLargeOrderAmounts = function (price, targetTokenDecimals, stableTokenDecimals, orderSizeLimit = MAX_ORDER_AMOUNT) {
+    let targetTokenAmount = orderSizeLimit.clone()
     let stableTokenAmount = getOutputAmountFromPrice(price, targetTokenAmount, targetTokenDecimals, stableTokenDecimals)
     if (stableTokenAmount.gt(targetTokenAmount)) {
```

Comment:
I was also expecting that this multiplicative inverse logic would no longer be a part of the function, but rather the exact price be specified by the function caller.